gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1131,14 +1131,27 @@ void gnc_file_ofx_import (GtkWindow *parent)
                                                           NULL, NULL);
             if (account)
             {
-                // Grab the balance value and date from the statement and open a reconcile window for this account.
-                gnc_numeric value = double_to_gnc_numeric (-statement->ledger_balance, xaccAccountGetCommoditySCU (account), GNC_HOW_RND_ROUND_HALF_UP);
-                recnWindowWithBalance (GTK_WIDGET (parent),
-                                       account,
-                                       value,
-                                       statement->ledger_balance_date);
+                if (statement->ledger_balance_valid)
+                {
+                    // The balance needs to be corrected for credit-card and loan accounts.
+                    float sign = 1;
+                    switch(xaccAccountGetType (account))
+                    {
+                        case ACCT_TYPE_CREDIT:
+                        case ACCT_TYPE_LIABILITY:
+                            sign = -1;
+                        default:
+                            sign = 1;
```

Comment:
This is not comprehensive enough. There is a distinct lack of `xaccAccountTypeGetFundamental` function which converts gnucash's account-types to one of the 5 fundamental types -- acount/liability/income/expense/equity. This also affects, for example, budget totals which sum only `ACCT_TYPE_ASSET` and ignores `ACCT_TYPE_BANK` budget amounts. See the following snippet onwards: https://github.com/Gnucash/gnucash/blob/802306503057a8f4476acc830eaa0e1f777e4c86/gnucash/gnome/gnc-budget-view.c#L1197

I suspect adding the following to `Account.cpp` is easy enough?
```cpp
GNCAccountType
xaccAccountTypeGetFundamental (GNCAccountType t)
{
    switch (t)
    {
    case ACCT_TYPE_BANK:
    case ACCT_TYPE_STOCK:
    case ACCT_TYPE_MUTUAL:
    case ACCT_TYPE_CURRENCY:
    case ACCT_TYPE_CASH:
    case ACCT_TYPE_ASSET:
    case ACCT_TYPE_RECEIVABLE:
        return ACCT_TYPE_ASSET;
    case ACCT_TYPE_CREDIT:
    case ACCT_TYPE_LIABILITY:
    case ACCT_TYPE_PAYABLE:
        return ACCT_TYPE_LIABILITY;
    case ACCT_TYPE_INCOME:
        return ACCT_TYPE_INCOME;
    case ACCT_TYPE_EXPENSE:
        return ACCT_TYPE_EXPENSE;
    case ACCT_TYPE_EQUITY:
        return ACCT_TYPE_EQUITY;
    case ACCT_TYPE_TRADING:
    default:
        return ACCT_TYPE_NONE;
    }
}
```